### ISSUE TITLE
Add a way to create a `Pair` with an already created `Optional<>`

### ISSUE THREAD

**Issue by nick-dolgiy** [SpineEventEngine/core-java#1228]
Spine allows having an `Optional` as the second argument of `Pair` but the only way to create it is call `Pair.withNullable()` which allows the second argument to be `null`, but there is no option to create a `Pair` if I have a message that is already wrapped in `Optional`.

I see the method signature like this:
``` Java
public static  <M extends Message, O extends Optional<? extends Message>> Pair<M, O> withOptional(M a, O b)
```

The resulting API will be:
``` Java
Optional<Message> optional = Optional.of(...);

Pair<Message, Optional<Message>> = Pair.withOptional(message, optional);
```

**Comment by serhii-lekariev** [comment]
Reassigned to @armiol.

**Comment by alexander-yevsyukov** [comment]
Closed by #1296.
